### PR TITLE
feat(core): roles crud

### PR DESCRIPTION
**Pull request by wangsijie**
## Summary

Add CRUD endpoints for basic role.

## Testing

UTs

### CODE REVIEW

File: packages/schemas/tables/usersroles.sql [hunk 1]
```diff
@@ -0,0 +1,5 @@
+create table users_roles (
```
Comment (gao-sun) [inline packages/schemas/tables/usersroles.sql:1]: same for the filename? `users_roles.sql`

File: packages/core/src/routes/role.test.ts [hunk 2]
```diff
@@ -19,4 +30,38 @@ describe('role routes', () => {
...
+    findRoleById.mockResolvedValueOnce(mockRole);
+    const response = await roleRequester
+      .patch(`/roles/${mockRole.id}`)
+      .send({ description: 'new' });
+    expect(response.status).toEqual(200);
+    expect(response.body).toEqual({
+      ...mockRole,
+      description: 'new',
+    });
+  });
```
Comment (darcyYe) [inline packages/core/src/routes/role.test.ts:60]: maybe should add the case that the role name you want to change has already been occupied.

File: packages/core/src/routes/role.ts [hunk 3]
```diff
@@ -1,11 +1,101 @@
...
+    koaGuard({
+      body: Roles.createGuard.omit({ id: true }),
+    }),
+    async (ctx, next) => {
+      const {
+        body,
+        body: { name },
+      } = ctx.guard;
+
+      assertThat(!(await hasRoleWithName(name)), 'role.name_in_use');
```
[the remark this answers is not in the document]
Comment (wangsijie) [reply packages/core/src/routes/role.ts]: Both are OK?

File: packages/schemas/tables/usersroles.sql (hunk 1 above)
Comment (wangsijie) [reply packages/schemas/tables/usersroles.sql]: discussed offline

File: packages/core/src/routes/role.test.ts (hunk 2 above)
Comment (wangsijie) [reply packages/core/src/routes/role.test.ts:63]: added

File: packages/phrases/src/locales/de/errors.ts [hunk 4]
```diff
@@ -175,6 +175,9 @@ const errors = {
   log: {
     invalid_type: 'Der Log Typ ist ungültig.',
   },
+  role: {
+    name_in_use: 'The name is already in use', // UNTRANSLATED
```
Comment (charIeszhao) [inline packages/phrases/src/locales/de/errors.ts:179]: nit: The `role` name is already in use

File: packages/core/src/routes/role.ts [hunk 5]
```diff
@@ -1,11 +1,101 @@
...
+    }),
+    async (ctx, next) => {
+      const {
+        body,
+        body: { name },
+        params: { id },
+      } = ctx.guard;
+
+      await findRoleById(id);
+      assertThat(!name || !(await findRoleByRoleName(name)), 'role.name_in_use');
```
Comment (charIeszhao) [inline packages/core/src/routes/role.ts:79]: We should exclude the current role name when calling `findRoleByRoleName` in PATCH request

File: packages/phrases/src/locales/de/errors.ts (hunk 4 above)
Comment (wangsijie) [reply packages/phrases/src/locales/de/errors.ts:179]: updated

File: packages/core/src/routes/role.ts (hunk 5 above)
Comment (wangsijie) [reply packages/core/src/routes/role.ts]: Thanks, updated

**Review by charIeszhao — APPROVED** [review]

**Pull request merged**